File: src/main/java/sn/brasilburger/Entity/LivraisonAffectation.java
package sn.brasilburger.Entity;

import sn.brasilburger.Entity.Enum.StatutLivraison;

public class LivraisonAffectation {
    private int id;
    private int commandeId;
    private int livreurId;
    private int zoneId;
    private StatutLivraison statut;

    public LivraisonAffectation() {
    }

    public LivraisonAffectation(int id, int commandeId, int livreurId, int zoneId, StatutLivraison statut) {
        this.id = id;
        this.commandeId = commandeId;
        this.livreurId = livreurId;
        this.zoneId = zoneId;
        this.statut = statut;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getCommandeId() {
        return commandeId;
    }

    public void setCommandeId(int commandeId) {
        this.commandeId = commandeId;
    }

    public int getLivreurId() {
        return livreurId;
    }

    public void setLivreurId(int livreurId) {
        this.livreurId = livreurId;
    }

    public int getZoneId() {
        return zoneId;
    }

    public void setZoneId(int zoneId) {
        this.zoneId = zoneId;
    }

    public StatutLivraison getStatut() {
        return statut;
    }

    public void setStatut(StatutLivraison statut) {
        this.statut = statut;
    }
}

File: CSharp/Models/LivraisonAffectation.cs
namespace Models
{
    using System.ComponentModel.DataAnnotations.Schema;
    public class LivraisonAffection
    {
        [Column("id")]
        public long Id { get; set; }
        [Column("commande_id")]
        public long CommandeId { get; set; }
        [Column("livreur_id")]
        public long LivreurId { get; set; }
        [Column("zone_id")]
        public long ZoneId { get; set; }
        [Column("statut")]
        public StatutLivraison Statut { get; set; }
    }
    public enum StatutLivraison { EN_ATTENTE, EN_COURS, TERMINEE }

}